### PR TITLE
bpo-38480: resource.setrlimit() should raise PermissionError

### PR DESCRIPTION
**Pull request by giampaolo**
See: https://bugs.python.org/issue38480

https://bugs.python.org/issue38480

### CODE REVIEW

**Review by zhangyangyu — CHANGES_REQUESTED** [review]
The whatsnew and changelog are wrong. It's `ValueError` and `setrlimit`. And the document of `setrlimit` should also be updated. Currently it says: raises ValueError if a process tries to raise its hard limit. I think we need a versionchanged in 3.9 documents when a not super user do this PermissionError is raised.

File: Doc/whatsnew/3.9.rst [hunk 1]
```diff
@@ -272,6 +272,9 @@ Changes in the Python API
 * The :mod:`venv` activation scripts no longer special-case when
   ``__VENV_PROMPT__`` is set to ``""``.
 
+* :func:`resource.setrusage` now raises :exc:`PermissionError` instead of
+  :exc:`KeyError` if the user doesn't have privileges.
```
Comment (pppery) [inline Doc/whatsnew/3.9.rst]:
```suggestion
  :exc:`ValueError` if the user doesn't have privileges.
```

**Comment by csabella** [comment]
@giampaolo, please take a look at the code review.  Thanks!

**Comment by giampaolo** [comment]
@zhangyangyu could you please take one last look?

**Review by zhangyangyu — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2019-10-16-00-07-21.bpo-38480.Fe6PqI.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+resource.setrusage() now raises PermissionError instead of KeyError  if the
```
Comment (zhangyangyu) [inline Misc/NEWS.d/next/Library/2019-10-16-00-07-21.bpo-38480.Fe6PqI.rst:1]: it's `setrlimit`

**Review by zhangyangyu — CHANGES_REQUESTED** [review]

**Pull request closed without merging**